src/document.rs

```diff
@@ -570,38 +569,20 @@ impl Document {
         Ok(images)
     }
 
-    pub fn decode_text(encoding: Option<&str>, bytes: &[u8]) -> String {
+    pub fn decode_text(encoding: Option<&Encoding>, bytes: &[u8]) -> Result<String> {
         if let Some(encoding) = encoding {
-            info!("{}", encoding);
-            match encoding {
-                "StandardEncoding" => bytes_to_string(encodings::STANDARD_ENCODING, bytes),
-                "MacRomanEncoding" => bytes_to_string(encodings::MAC_ROMAN_ENCODING, bytes),
-                "MacExpertEncoding" => bytes_to_string(encodings::MAC_EXPERT_ENCODING, bytes),
-                "WinAnsiEncoding" => bytes_to_string(encodings::WIN_ANSI_ENCODING, bytes),
-                "PDFDocEncoding" => bytes_to_string(encodings::PDF_DOC_ENCODING, bytes),
-                "UniGB-UCS2-H" | "UniGB−UTF16−H" => UTF_16BE.decode(bytes).0.to_string(),
-                "Identity-H" => "?Identity-H Unimplemented?".to_string(), // Unimplemented
-                _ => String::from_utf8_lossy(bytes).to_string(),
-            }
+            info!("Decoding text with {:#?}", encoding);
+            encoding.bytes_to_string(bytes)
         } else {
-            bytes_to_string(encodings::STANDARD_ENCODING, bytes)
+            Ok(bytes_to_string(&encodings::STANDARD_ENCODING, bytes))
         }
     }
 
-    pub fn encode_text(encoding: Option<&str>, text: &str) -> Vec<u8> {
+    pub fn encode_text(encoding: Option<&Encoding>, text: &str) -> Vec<u8> {
         if let Some(encoding) = encoding {
-            match encoding {
-                "StandardEncoding" => string_to_bytes(encodings::STANDARD_ENCODING, text),
-                "MacRomanEncoding" => string_to_bytes(encodings::MAC_ROMAN_ENCODING, text),
-                "MacExpertEncoding" => string_to_bytes(encodings::MAC_EXPERT_ENCODING, text),
-                "WinAnsiEncoding" => string_to_bytes(encodings::WIN_ANSI_ENCODING, text),
-                "PDFDocEncoding" => string_to_bytes(encodings::PDF_DOC_ENCODING, text),
-                "UniGB-UCS2-H" | "UniGB−UTF16−H" => encodings::encode_utf16_be(text).to_vec(),
-                "Identity-H" => vec![], // Unimplemented
-                _ => text.as_bytes().to_vec(),
-            }
+            encoding.string_to_bytes(text)
         } else {
-            string_to_bytes(encodings::STANDARD_ENCODING, text)
+            string_to_bytes(&encodings::STANDARD_ENCODING, text)
         }
     }
 }
```

Comment:
I agree, according to spec the StandardEncoding is a fallback only for Type 1 fonts in only certain circumstances, and is rarely used.  By leaving it I wanted to maintain current behavior.

Anyway it will be cleaned up if we remove the Option.